src/Microsoft.OData.Client/ClientEdmModel.cs

```diff
@@ -328,7 +328,7 @@ private static Type[] GetTypeHierarchy(Type type, out PropertyInfo[] keyProperti
                 {
                     hierarchy.Insert(0, type);
                 }
-                while ((type = c.PlatformHelper.GetBaseType(type)) != null);
+                while ((type = c.PlatformHelper.GetBaseType(type)) != null && type != typeof(object));

```

Comment:
>&& type != typeof(object) [](http://example.com/codeflow?start=75&length=25)

should we put this check first? is it cheaper? more likely to be true?

Reply:
@mikepizzo Because this is a do...while loop, we want to check if `type` after it has been assigned. The assignment happens in the first expression: `type = c.PlatformHelper.GetBaseType(type)`